skbio/diversity/alpha/_base.py

```diff
@@ -848,6 +850,51 @@ def osd(counts):
     return observed_otus(counts), singles(counts), doubles(counts)
 
 
+@experimental(as_of="0.4.0-dev")
+def pielou_e(counts):
+    r"""Calculate Pielou's Evenness index J'.
+
+    Pielou's Evenness is defined as:
+
+    .. math::
+
+       J' = \frac{(H)}{\ln(S)}
+
+    where :math:`H` is the Shannon-Wiener entropy of counts and :math:`S` is
+    the number of OTUs in the sample.
+
+    Parameters
+    ----------
+    counts : 1-D array_like, int
+        Vector of counts.
+
+    Returns
+    -------
+    double
+        Pielou's Evenness.
+
+    See Also
+    --------
+    shannon
+    heip_e
+
+    Notes
+    -----
+    The implementation here is based on the description given by Carlo Heip
+    [1]_ and is also described on Wikipedia [2]_. It is similar to Heip's
+    evenness [1]_.
+
+    References
+    ----------
+    .. [1] Heip, C. 1974. A new index measuring evenness. J. Mar. Biol. Ass.
+       UK., 54, 555-557.
+    .. [2] https://en.wikipedia.org/wiki/Species_evenness
+
+    """
+    counts = _validate_counts_vector(counts)
+    return shannon(counts) / np.log(observed_otus(counts))
```

Comment:
No problem, we should probably change the default logarithm base to `e` in `shannon` to avoid issues like this in the future (since there are several diversity metrics that use Shannon internally but assume natural log). Definitely doesn't need to be addressed in this PR.

Here's a few options to get unbiased expected values to use in your unit tests:

- [free online calculator](https://www.easycalculation.com/statistics/shannon-wiener-diversity.php)

- [this site](http://classroom.synonym.com/calculate-species-evenness-2851.html) runs through the calculations by hand with a small example, so you could add their example dataset to the unit tests and include a link (as a comment) to where the expected value came from

- perhaps the Heip paper you're linking to has an example dataset with expected values for Pielou's evenness?

- [SDR-IV](http://www.pisces-conservation.com/index.html?softdiversity.html$softwaremenu.html) has a free demo but probably only runs on Windows